http/server.ts

```diff
@@ -8,6 +8,8 @@ import { TextProtoReader } from "../textproto/mod.ts";
 import { STATUS_TEXT } from "./http_status.ts";
 import { assert } from "../testing/asserts.ts";
 
+export let listener: Deno.Listener;
```

Comment:
This makes the listener global - which doesn’t allow for there to be multiple listeners. You’ll have to find another way to expose it.

Reply:
could I do like the following pseudo code? If you use another method, it may cause damaging changes. @ry 
```ts
const listeners = new Map<string, Deno.Listener>();
...
export async function* serve(
  addr: string
): AsyncIterableIterator<ServerRequest> {
  const listener = listen("tcp", addr);
  listeners.set(addr, listener);
  ...
}
export function getListener(addr: string) {
  return listeners.get(addr);
}
```

Reply:
How about exporting a wrapper function?

```ts
export function serve(addr: string) {
  const listener = listen("tcp", addr)
  const iterator = (async function *() {
    // ...
  })()
  iterator.listener = listener
  return iterator
}

Reply:
agree with @j-f1 . I think this is the cleanest way to do it.